### ISSUE TITLE
Minimum data required for a functional go-bot

### ISSUE THREAD

**Issue by vaibhavgeek** [deepmipt/DeepPavlov#509]
I am going through http://docs.deeppavlov.ai/en/master/skills/go_bot.html#dialogs and am wondering what is the minimum data required for following the method 2, that is providing custom dataset reader and data for training, testing and validation. 

I understand it would be varying for use-case to use-case but can you provide a ball park figure that you seen in your experience. Looking forward for your response. 

Warm Regards,

**Comment by vikmary** [comment]
Hi Vaibhav,
So the question is which way is faster: to replace train, valid and test files with yours or to write your own `DatasetReader`?

**Comment by vaibhavgeek** [comment]
Hi @vikmary,
I am sorry to sound extremely vague. I am looking forward to create a custom goal oriented chatbot with help deeppavlov library. I have to label the data by myself and hence creating the train.txt, valid.txt and test.txt (similar to dstc2) is huge task for me and not feasible. 

1. Can you please guide on how to create a custom goal oriented chatbot? (A referral bot config json file for custom bot would really help).

2. I have seen the documentation and various issues in this repo. Mostly all of them require huge amount of data labelling. Is that true? 

3. What is the minimum data required to build a functional goal oriented chatbot?

I thank you for responding to my issue this quick. A lot of people are looking forward to build their custom goal oriented chatbot, it would be great if you can provide with a tutorial for the same. 

Warm Regards,
Vaibhav Maheshwari

**Comment by vikmary** [comment]
Hi @vaibhavgeek 

I would recommend three approaches sorted by ascending amount of training data:

0. **Rules.** You always have the option, no data is required.

1. If you have at least 10 examples per intent, then it is worth trying **Intent Classifier + rules**. This is the simplest model: to train a classified of predefined intents. So input utterance is labeled with one of the classes and then hand-written rules decide what to respond (your responses can deterministic or have several paraphrases). DeepPavlov contains a good [intent classifier](http://docs.deeppavlov.ai/en/master/components/classifiers.html) to start with.

2. DeepPavlov [**go_bot**](http://docs.deeppavlov.ai/en/master/skills/go_bot.html) based on Hybrid Code Networks. For details see the paper. The advancement is that what to respond is decided by a learned recursive neural network. But still responses are deterministic. I used the DSTC2 dataset to train the model, it contains 1000 dialogs in training set. The required amount of training data strongly depends on the complexity of dialogs in your domain (number of possible decision tree branches).

To be able to output restaurant info correctly, anonymize slot mentions in your training data: in input utterances and output utterances (for example, "Starbucks is on Red Square street, 1" -> "<restaurant_name> is on <restaurant_address>"). To anonymize restaurant slots in user utterances during inference you will need a Named Entity Recognizer ("I want cheap food" -> NER model -> {"food": "cheap"} -> "I want <restaurant_pricerange>").  Our [NER model](http://docs.deeppavlov.ai/en/master/components/ner.html) works great in cases when you have 100+ examples per slot.

**Comment by vaibhavgeek** [comment]
I have a small doubt. 

>  The required amount of training data strongly depends on the complexity of dialogs in your domain (number of possible decision tree branches).

How much?

1000 dialogues is way too much data as a training data for a decision tree implementation of the type DSTC2 (in my opinion). I think with database and templates as input data , a smaller train can do the job (or train file can be generated somehow). I want test this hypothesis by myself so I would be experimenting a bit on this source code. I will let this community know of my results. 

Thank you for taking out time to answer my queries.

**Comment by vaibhavgeek** [comment]
Was the database restro.sqlite generated through dstc2 training data?
